File: backend/app/schemas/clip.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from typing import Optional


class ClipResponse(BaseModel):
    id: UUID
    media_id: str
    title: str
    season_episode: Optional[str] = None
    start_time_ms: int
    end_time_ms: int
    duration_ms: int
    composite_score: float
    genre_tags: list[str] = []
    actors: list[str] = []
    director: Optional[str] = None
    decade: Optional[str] = None
    mood_tags: list[str] = []
    thumbnail_url: Optional[str] = None
    stream_url: str
    created_at: datetime

    model_config = {"from_attributes": True}


class FeedResponse(BaseModel):
    clips: list[ClipResponse]
    offset: int
    limit: int
    has_more: bool

File: ios/BYETZ/Models/Clip.swift
import Foundation

struct Clip: Codable, Identifiable {
    let id: UUID
    let mediaId: String
    let title: String
    let seasonEpisode: String?
    let startTimeMs: Int
    let endTimeMs: Int
    let durationMs: Int
    let compositeScore: Double
    let genreTags: [String]
    let actors: [String]
    let director: String?
    let decade: String?
    let moodTags: [String]
    let thumbnailUrl: String?
    let streamUrl: String
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case mediaId = "media_id"
        case title
        case seasonEpisode = "season_episode"
        case startTimeMs = "start_time_ms"
        case endTimeMs = "end_time_ms"
        case durationMs = "duration_ms"
        case compositeScore = "composite_score"
        case genreTags = "genre_tags"
        case actors
        case director
        case decade
        case moodTags = "mood_tags"
        case thumbnailUrl = "thumbnail_url"
        case streamUrl = "stream_url"
        case createdAt = "created_at"
    }
}

struct FeedResponse: Codable {
    let clips: [Clip]
    let hasMore: Bool

    enum CodingKeys: String, CodingKey {
        case clips
        case hasMore = "has_more"
    }
}
